std/format.d

```diff
@@ -6549,3 +6549,14 @@ unittest
     assert(sformat(buf[], "%s %s %s", "c"c, "w"w, "d"d) == "c w d");
     });
 }
+
+/*****************************
+ * The .ptr is unsafe because it could be dereferenced and the length of the array may be 0.
+ * Returns:
+ *      the difference between the starts of the arrays
+ */
+@trusted private pure nothrow @nogc
+    size_t arrayPtrDiff(const void[] array1, const void[] array2)
```

Comment:
Shouldn't this return `ptrdiff_t` or at least assert that `array1.ptr >= array2.ptr`?